src/query/tsdb/remote/codecs.go

```diff
@@ -313,13 +293,49 @@ func decodeTagMatchers(rpcMatchers *rpc.TagMatchers) (models.Matchers, error) {
 	return models.Matchers(matchers), nil
 }
 
+func decodeFanoutOption(opt rpc.FanoutOption) (storage.FanoutOption, error) {
+	switch opt {
+	case rpc.FanoutOption_DEFAULT_OPTION:
+		return storage.FanoutDefault, nil
+	case rpc.FanoutOption_FORCE_DISABLED:
+		return storage.FanoutForceDisable, nil
+	case rpc.FanoutOption_FORCE_ENABLED:
+		return storage.FanoutForceEnable, nil
+	}
+
+	return storage.FanoutDefault,
+		fmt.Errorf("Unknown fanout option for proto encoding: %v\n", opt)
```

Comment:
Same comments on just `return 0, errorWhatever` and also using `fmt.Errorf("unknown ...: %v")` (uncapitalize the unknonwn and remove the `\n`).